pkg/controller/oneagent/istio_test.go

```diff
@@ -19,34 +25,43 @@ var (
 	namespace  = "dynatrace"
 )
 
-func TestReconcileOneAgent_ReconcileIstio(t *testing.T) {
-	oa := newOneAgentSpec()
-	oa.ApiUrl = testAPIUrl
-	oa.Tokens = "token_test"
-	oa.EnableIstio = true
-	dynatracev1alpha1.SetDefaults_OneAgentSpec(oa)
+func TestReconcileOneAgent_CreateIstioObjects(t *testing.T) {
 
-	reconcileOA, client := setupReconciler(t, oa)
+	buffer := bytes.NewBufferString("{\"apiVersion\":\"networking.istio.io/v1alpha3\",\"kind\":\"VirtualService\",\"metadata\":{\"clusterName\":\"\",\"creationTimestamp\":\"2018-11-26T03:19:57Z\",\"generation\":1,\"name\":\"test-virtual-service\",\"namespace\":\"istio-system\",\"resourceVersion\":\"1297970\",\"selfLink\":\"/apis/networking.istio.io/v1alpha3/namespaces/istio-system/virtualservices/test-virtual-service\",\"uid\":\"266fdacc-f12a-11e8-9e1d-42010a8000ff\"},\"spec\":{\"gateways\":[\"test-gateway\"],\"hosts\":[\"*\"],\"http\":[{\"match\":[{\"uri\":{\"prefix\":\"/\"}}],\"route\":[{\"destination\":{\"host\":\"test-service\",\"port\":{\"number\":8080}}}],\"timeout\":\"10s\"}]}}\n")
 
-	// mocking the request
-	req := reconcile.Request{
-		NamespacedName: types.NamespacedName{
-			Name:      name,
-			Namespace: namespace,
-		},
+	vs := istioV1alpha3.VirtualService{}
+	err := json.Unmarshal(buffer.Bytes(), &vs)
+
+	ic := fakeIstio.NewSimpleClientset(&vs)
+
+	vsList, err := ic.NetworkingV1alpha3().VirtualServices("istio-system").List(metav1.ListOptions{})
+	if err != nil {
+		t.Errorf("Failed to create VirtualService in %s namespace: %s", namespace, err)
 	}
-	_, err := reconcileOA.Reconcile(req)
+	if len(vsList.Items) == 0 {
+		t.Error("Expected items, got nil")
+	}
+	t.Logf("list of istio object %v", vsList.Items)
+}
+
+func TestReconcileOneAgent_BuildDynatraceVirtualService(t *testing.T) {
+	os.Setenv(k8sutil.WatchNamespaceEnvVar, namespace)
+
+	buffer := istio.BuildVirtualService("dt-vs", "ENVIRONMENTID.live.dynatrace.com", 443, "https")
+	vs := istioV1alpha3.VirtualService{}
+	err := json.Unmarshal(buffer, &vs)
 	if err != nil {
-		t.Fatalf("error reconciling: %v", err)
+		t.Errorf("Failed to marshal json %s", err)
 	}
-	virtualService := getGVK(client, istio.VirtualServiceGVK)
-	if virtualService == nil {
-		t.Error("no istio virtual services objects formed")
+	ic := fakeIstio.NewSimpleClientset(&vs)
+	vsList, err := ic.NetworkingV1alpha3().VirtualServices(namespace).List(metav1.ListOptions{})
+	if err != nil {
+		t.Errorf("Failed to create VirtualService in %s namespace: %s", namespace, err)
 	}
-	serviceEntry := getGVK(client, istio.ServiceEntryGVK)
-	if serviceEntry == nil {
-		t.Error("no istio objects for service entry")
+	if len(vsList.Items) == 0 {
+		t.Error("Expected items, got nil")
 	}
+	t.Logf("list of istio object %v", vsList.Items)
 }
 
 func TestReconcileOneAgent_ReconcileIstioViaDynatraceClient(t *testing.T) {
```

Comment:
Is it possible to check for actually created objects? We're only reconciling twice and checking whether if the function was successful or not, which doesn't say much.

Reply:
Not without the actual mock flow for rest client. Can reserve it for housekeeping?

Reply:
OK, we can take it there.